nixos/modules/services/networking/icmptx.nix

```diff
@@ -0,0 +1,199 @@
+{
+  config,
+  lib,
+  pkgs,
+  utils,
+  ...
+}:
+let
+  cfg = config.services.icmptx;
+in
+{
+  options = {
+    services.icmptx =
+      let
+        commonOptions = {
+          tun = lib.mkOption {
+            type = lib.types.str;
+            default = "tun0";
```

Comment:
Why do we hardcode tun0 and not icmptx0?

Reply:
Because the ICMPTX program does not allow changing the interface name and will always use `tun0` if available.

Reply:
I added this option in case changing the name becomes possible in the future.